tests/phpunit/tests/post/nav-menu.php

```diff
@@ -265,6 +265,77 @@ public function test_update_menu_item_cache_primes_terms() {
 		$this->assertSameSets( array( $term_id ), $last[1], '_prime_term_caches() was not executed.' );
 	}
 
+
+	/**
+	 * @ticket 55620
+	 * @covers ::update_menu_item_cache
+	 */
+	public function test_wp_get_nav_menu_items_cache_primes_posts() {
+		$post_ids     = self::factory()->post->create_many( 3 );
+		$menu_nav_ids = array();
+		foreach ( $post_ids as $post_id ) {
+			$menu_nav_ids[] = wp_update_nav_menu_item(
+				$this->menu_id,
+				0,
+				array(
+					'menu-item-type'      => 'post_type',
+					'menu-item-object'    => 'post',
+					'menu-item-object-id' => $post_id,
+					'menu-item-status'    => 'publish',
+				)
+			);
+		}
+
+		wp_cache_delete_multiple( $menu_nav_ids, 'posts' );
+		wp_cache_delete_multiple( $post_ids, 'posts' );
+		$action = new MockAction();
+		add_filter( 'update_post_metadata_cache', array( $action, 'filter' ), 10, 2 );
+
+		wp_get_nav_menu_items( $this->menu_id );
+
+		$args = $action->get_args();
+		$this->assertSameSets( $menu_nav_ids, $args[0][1], '_prime_post_caches() was not executed.' );
+		$this->assertSameSets( $post_ids, $args[1][1], '_prime_post_caches() was not executed.' );
```

Comment:
```suggestion
		$this->assertSameSets( $menu_nav_ids, $args[0][1], '_prime_post_caches() was not executed for navigation menu items.' );
		$this->assertSameSets( $post_ids, $args[1][1], '_prime_post_caches() was not executed for linked post objects.' );
```

I think this test is flakey, they only fail on trunk due to the order of operations:

* db queries: 6 on both trunk and this branch
* posts cached: yes. runs first on trunk, second on this branch
* nav items cached: yes. Runs second on trunk, first on this branch